src/model/webhook.rs

```diff
@@ -191,28 +194,32 @@ impl WebhookChannel {
     ///
     /// Returns [`Error::Http`] if the channel retrieval request failed.
     #[inline]
-    pub async fn to_channel(self, cache_http: impl CacheHttp) -> Result<Channel> {
+    pub async fn to_channel(self, cache_http: impl CacheHttp) -> Result<GuildChannel> {
         #[cfg(feature = "cache")]
         {
             if let Some(cache) = cache_http.cache() {
                 if let Some(channel) = cache.channel(self.id) {
-                    return Ok(channel);
+                    return Ok(channel.clone());
                 }
             }
         }
 
         let channel = cache_http.http().get_channel(self.id).await?;
-
-        #[cfg(all(feature = "cache", feature = "temp_cache"))]
-        {
-            if let Some(cache) = cache_http.cache() {
-                if let Channel::Guild(guild_channel) = &channel {
-                    cache.temp_channels.insert(guild_channel.id, guild_channel.clone());
+        match channel {
+            Channel::Guild(guild_channel) => {
+                #[cfg(all(feature = "cache", feature = "temp_cache"))]
+                {
+                    if let Some(cache) = cache_http.cache() {
+                        cache
+                            .temp_channels
+                            .insert(guild_channel.id, Arc::new(guild_channel.clone()));
+                    }
                 }
-            }
-        }
 
-        Ok(channel)
+                Ok(guild_channel)
+            },
+            other => Err(Error::Model(ModelError::NonGuildWebhookChannel(Box::new(other)))),
```

Comment:
```suggestion
            other => Err(Error::Model(ModelError::InvalidChannelType)),
```
Couldn't we just use `InvalidChannelType` here and call it a day? If it's not possible to make a webhook for a private channel, it's not worth creating an error variant that will never realistically be constructed at runtime.